sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -172,8 +172,7 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
       // We can't reuse the broadcast because the join type doesn't support broadcast,
       // and doing DPP means running an extra query that may have significant overhead.
       // We need to make sure the pruning side is very big so that DPP is still worthy.
-      canBroadcastBySize(otherPlan, conf) &&
```

Comment:
I'm not very sure about this. We still need to collect the build side data in the driver, and it shouldn't be too big to avoid driver OOM.

Reply:
As this is not a broadcast join case, and we don't need broadcast it, if we really need a size check, we should use another config for the purpose. `autoBroadcastJoinThreshold` doesn't look like a good option because its meaning doesn't fit here.

Reply:
I added one config to set a threshold for this query collecting.

Reply:
cc @maryannxue FYI

Reply:
I'm thinking could we remove it to make `PartitionPruning` more clear. This is because this approach has 2 disadvantages:
1. Disable `spark.sql.optimizer.dynamicPartitionPruning.reuseBroadcastOnly`.
2. The filtering side will be evaluated 2 times. For example: https://github.com/apache/spark/pull/29726#issuecomment-780266596


Instead, we can use bloom filter join pruning. It has 2 advantages:
1. Do not need consider broadcast.
2. Reuse shuffle exchange to avoid the filtering side evaluate 2 times. For example: https://github.com/apache/spark/pull/33914#issuecomment-913120675

Reply:
Remove what? You mean not trigger DPP at all if it is not a broadcast join?